src/app/mod.rs

```diff
@@ -1023,6 +1023,37 @@ impl<'a, 'b> App<'a, 'b> {
         self.p.gen_completions(for_shell, out_dir.into());
     }
 
+
+    /// Generate a completions file for a specified shell at runtime.  Until `cargo install` can
+    /// install extra files like a completion script, this may be used e.g. in a command that
+    /// outputs the contents of the completion script, to be redirected into a file by the user.
+    ///
+    /// # Examples
+    ///
+    /// TODO: can this somehow reference `cli.rs` from the example above or otherwise share that
+    /// part of the example?
+    ///
+    /// ```ignore
+    /// src/main.rs
+    /// mod cli;
+    /// use std::io;
+    ///
+    /// fn main() {
+    ///     let matches = cli::build_cli().get_matches();
+    ///
+    ///     if matches.is_present("generate-bash-completions") {
+    ///         cli::build_cli().gen_completions_to("myapp", Shell::Bash, &mut io::stdout());
+    ///     }
+    ///
+    ///     // normal logic continues...
+    /// }
+    ///
+    /// ```
+    pub fn gen_completions_to<W: Write, S: Into<String>>(&mut self, bin_name: S, for_shell: Shell, buf: &mut W) {
```

Comment:
i notice that in the existing `gen_completions`, the trait bounds (not sure about terminology, the `<W: ..., S: ...>` part has the variables in the reverse order to the function parameters. is this arbitrary? are there conventions?